### PR TITLE
Adds display: contents

### PR DESCRIPTION
**Pull request by vwallen**
This would replace an existing draft PR that appears inactive: https://github.com/web-platform-dx/web-features/pull/1131

The support story here is messy: BCD shows high support, but Can I Use shows partial support across the board due to buggy implementations or significant accessibility problems (for example, removing `<button>` elements from focus navigation). 

The `css.properties.display.contents.contents_unusual` refers to default behavior for elements where `display: contents` would result in incomplete, non-semantic, or non-nonsensical behavior. This part of the spec doesn't provide any guidance as to how this should affect accessibility.

Closes #1131.

### CODE REVIEW

File: features/display-contents.yml
Comment (vwallen) [inline features/display-contents.yml:4]: Disagreement between Can I Use and BCD regarding level of support (versions match)

**Comment by vwallen** [comment]
@captainbrosset I saw your comment on the other PR (https://github.com/web-platform-dx/web-features/pull/1131).

How does this look to you?

File: features/display-contents.yml
Comment (captainbrosset) [inline features/display-contents.yml:1]: The accessibility part is interesting here. I did some research.

My understanding is that the spec didn't originally made it clear to implementers what should happen, and so, what ended up happening is that elements that should be focusable were no longer focusable when set to display:contents. That's because these elements would no longer generate a box in the layout for themselves, and therefore, the browser didn't have anywhere to put the focus on.

This became an accessibility issue, reported [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1791648), and [here](https://bugs.chromium.org/p/chromium/issues/detail?id=1366037).
So much, that web industry leaders told devs not to use display:contents at all.

Because not everyone knew about this, an [interop focus area](https://github.com/web-platform-tests/interop/issues/568) was proposed last year to fix this, and got accepted in the Interop 24 cycle.
Unfortunately, it seems like that focus area got scoped down, and [the description of the focus area](https://github.com/web-platform-tests/interop/blob/main/2024/README.md#accessibility) only says that a new test should be added, but doesn't require implementers to pass the test. [The test now exists](https://wpt.fyi/results/css/css-display/focus).

Turns out this test now passes on chromium browsers, and that's thanks to [David Baron implementing the fix in January 2024](https://chromium-review.googlesource.com/c/chromium/src/+/3910374).

The status is the following now:

- On Chromium-based browsers, elements that have a display:contents property **are** now focusable (e.g. tabbing through a page will go through them, `document.activeElement` is updated accordingly, and calling `el.focus()` does work).
  - However, note that no focus ring appear yet. This is considered as a separate issue that still needs discussion.
- On Safari and Firefox, those elements **are not** focusable, so the accessibility problem remains.

My recommendation:

We decided early on in the project that accessibility was not (yet) going to impact baseline status. Also, we don't have the concept of partial support like caniuse (which shows the feature as supported, but with a note).

So I would recommend making this a baseline:high feature, computed from `css.properties.display.contents`, because that's the property devs can use today, and it does the job, and my understanding is that it doesn't impact accessibility negatively if you don't use it on focusable elements.
But also, keep that "may lose semantic or accessibility features" part in your description.

The other option would be to add a new BCD key named something like `css.properties.display.contents.focusable` which would be true for chromium, and false for others. And then, we'd potentially create two features:

1. `Inaccessible display:contents`, computed from `css.properties.display.contents`, baseline high.
2. `display:contents`, computed from `css.properties.display.contents.focusable`, not baseline.

@ddbeck to weight in on this. The latter option might seem weird, but because this is a current problem which devs are asked to think about, so having a feature about it isn't unreasonable. And, over time, those 2 features could be merged.

**Review by captainbrosset — CHANGES_REQUESTED** [review]

File: features/display-contents.yml
Comment (vwallen) [reply features/display-contents.yml:10]: Updated

File: features/display-contents.yml
Comment (ddbeck) [reply features/display-contents.yml:1]: > We decided early on in the project that accessibility was not (yet) going to impact baseline status. Also, we don't have the concept of partial support like caniuse (which shows the feature as supported, but with a note).

This isn't exactly right. We modified the Baseline doc (see https://github.com/web-platform-dx/web-features/pull/519) to say that assistive technology _outside the browser_ is out of scope. Making elements unfocusable and not showing focus states (which are stylable now)—all out of spec—seems in-bounds for Baseline (and BCD). We can and should capture this information.

> So I would recommend making this a baseline:high feature, computed from `css.properties.display.contents`, because that's the property devs can use today, and it does the job, and my understanding is that it doesn't impact accessibility negatively if you don't use it on focusable elements.

I don't agree that we should have two features for this (at least without https://github.com/web-platform-dx/web-features/issues/915). The current, surprising behavior of `display: contents` feels distinctly broken. It's a "feature" that, if used, breaks other features we describe (or will shortly) as widely available (such as `tabindex`).

Instead, I think we should (as you suggested) create a behaviorial BCD entry like `css.properties.display.contents.focusable_descendants` then set the support data based on the results of the WPT case.

Once that feature lands in BCD, we can then publish a single `display: contents` feature that depends on the new BCD key. We ought to omit the "…lose semantic" bit from the description, as that is flatly _not_ an intended or desirable behavior of the feature.

Right now, I think that means blocking this PR (or moving it draft status) or merging it into the `drafts` folder and picking this up again when the BCD work is done.

(As an aside, I only somewhat recently learned about `display: contents`, used it in a personal project and now—thanks to this discussion—I'm removing it. It silently broke things and that makes me sad.)

Comment (captainbrosset) [reply features/display-contents.yml:1]: +1 from me on going the BCD route for this, and marking this PR as draft in the meantime.

Comment (ddbeck) [reply features/display-contents.yml:1]: I've sent https://github.com/mdn/browser-compat-data/pull/24766. I'll update here if/when it merges in BCD.

File: features/display-contents.yml
Comment (ddbeck) [inline features/display-contents.yml:2]: Continuing from https://github.com/web-platform-dx/web-features/pull/1966/files#r1803434536

```suggestion
description: "The `display: contents` CSS declaration sets an element to display only its contents, not itself."
```

File: features/display-contents.yml
Comment (ddbeck) [inline features/display-contents.yml:7]: Continuing from https://github.com/web-platform-dx/web-features/pull/1966/files#r1803434536

Let's remove this:

```suggestion
```

File: features/display-contents.yml
Comment (ddbeck) [inline features/display-contents.yml:10]: Continuing from https://github.com/web-platform-dx/web-features/pull/1966/files#r1803434536

```suggestion
  - css.properties.display.contents.contents_unusual
  - css.properties.display.contents.focusable_elements
```

File: features/display-contents.yml
Comment (ddbeck) [reply features/display-contents.yml:1]: [css.properties.display.contents.focusable_elements](https://github.com/mdn/browser-compat-data/pull/24766) has merged, so we'll get it in the next BCD release (probably Friday or sooner). I'll add some comments suggesting what to do with this, specifically.

**Review by ddbeck — CHANGES_REQUESTED** [review]
I've made some suggestions to handle the accessibility issue. This is still blocked until the next BCD release, even if these suggestions are resolved.

**Review by ddbeck — APPROVED** [review]

**Pull request merged**
